extra/ldr-to-hdr.sh

```diff
@@ -0,0 +1,226 @@
+	echo "Outputs to CWD, with the HDR name being the MD5SUM"
+	echo "Required arguments:"
+	echo "-h [path] | HDRGen binary path"
+	echo "-r [path] | Radiance package binary path"
+	echo "-i [path/file wildcard] | Input image path/wildcard"
+	echo "-z [file] | Calibration Factoring amount (ie. 1.34)"
+	echo "-x [file] | Vignetting Config File"
+	echo "-c [file] | Neutral Density Filter File"
+	echo "-v [file] | Fisheye Correction File"
+	echo "-b [file] | Camera Response File"
+	echo "-n [file] | Camera Settings File"
+}
+
+# Check to see if no args were given
+
+if [[ -z $1 ]]; then
+	printhelp
+	exit 0
+fi
+
+# Argument Parsing
+
+# -h| HDRGEN path
+# -r| Radiance path
+# -i| Input images, can be regex or directory too.
+# ARUGMENTS ARE DESIGNED TO BE NEXT TO EACH OTHER ON THE KEYBOARD FOR EASE OF USE.
+# -z| Calibration Factoring amount
+# -x| Vignetting Config File
+# -c| Neutral Density Filter File
+# -v| Fisheye Correction File
+# -b| Camera Response File
+# -n| Camera Settings File
+
+while getopts "h:r:i:z:x:c:v:b:n:" arg; do
+  case $arg in
+    h|r) # Setting HDRGEN path or radiance path
+      export PATH+=":$OPTARG"
+      ;;
+    i) # Getting the input files
+      INPUT_FILES=("$OPTARG"*)
+      for file in $INPUT_FILES; do
+        if ! [[ -f "$file" ]]; then
+          echo "ERROR LOADING FILES: $file IS NOT A FILE.";
+	        echo "Other files attempting to load:"
+          printf '%s\n' "${INPUT_FILES[@]}"
+	        exit 1
+        fi
+      done;
+      ;;
+    z) # Calibration Factoring File
+      CONFIG_CALFAC_PATH=$OPTARG
+      ;;
+    x) # Vignetting Config File
+      CONFIG_VIG_PATH=$OPTARG
+      ;;
+    c) # Neutral Density Filter File
+      CONFIG_ND_PATH=$OPTARG
+      ;;
+    v) # Fisheye Correction File
+      CONFIG_FISHEYE_PATH=$OPTARG
+      ;;
+    b) # Camera Response File
+      CONFIG_CAMRESP_PATH=$OPTARG
+      ;;
+    n) # Camera Setting File
+      # Use regex magic to parse the file for all the information we need.
+      CONFIG_FISHEYE_DIAM=$(grep -Po "(?<=diameter <- ).*" $OPTARG || ggrep -Po "(?<=diameter <- ).*" $OPTARG )
+      CONFIG_CROP_X=$(grep -Po "(?<=xleft <- ).*" $OPTARG || ggrep -Po "(?<=xleft <- ).*" $OPTARG)
+      CONFIG_CROP_Y=$(grep -Po "(?<=ydown <- ).*" $OPTARG || ggrep -Po "(?<=ydown <- ).*" $OPTARG)
+      CONFIG_VA_V=$(grep -Po "(?<=-vv ).*" $OPTARG || ggrep -Po "(?<=-vv ).*" $OPTARG)
+      CONFIG_VA_H=$(grep -Po "(?<=-vh ).*" $OPTARG || ggrep -Po "(?<=-vh ).*" $OPTARG)
+      CONFIG_OUT_X=$(grep -Po "\d*(?=x\d*)" $OPTARG || ggrep -Po "\d*(?=x\d*)" $OPTARG)
+      CONFIG_OUT_Y=$(grep -Po "(?<=\dx)\d*" $OPTARG || ggrep -Po "(?<=\dx)\d*" $OPTARG)
+      ;;
+    *) # Unknown argument given
+      echo "Unknown argument given..."
+      echo "$OPTARG"
+      
+  esac
+done
+
+# Ensure all variable are set
+
+if [[ -z $INPUT_FILES ]]; then
+	echo "ERROR: No input files given, please give as path to all files."
+	exit 1
+fi
+if [[ -z $CONFIG_CALFAC_PATH ]]; then
```

Comment:
For some reason, when I try to run the script, it keeps letting me know the calibration file isn't there, even though I believe I've included it in the list of arguments. I can't figure out if I'm somehow entering them wrong, or if it's maybe just macOS not playing nice with this script.